consensus/src/payload_client.rs

```diff
@@ -11,16 +11,29 @@ use aptos_consensus_types::{
     request_response::{GetPayloadCommand, GetPayloadResponse},
 };
 use aptos_logger::prelude::*;
+use aptos_types::validator_txn::{
+    pool::{ValidatorTransactionFilter, ValidatorTransactionPoolClient},
+    ValidatorTransaction,
+};
 use fail::fail_point;
 use futures::{
     channel::{mpsc, oneshot},
     future::BoxFuture,
 };
-use std::time::{Duration, Instant};
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
 use tokio::time::{sleep, timeout};
 
 const NO_TXN_DELAY: u64 = 30; // TODO: consider moving to a config
 
+pub struct MixedPayloadClient {
+    pub validator_txn_enabled: bool,
+    pub validator_txn_pool_client: Arc<dyn ValidatorTransactionPoolClient>,
+    pub quorum_store_client: QuorumStoreClient,
```

Comment:
this can still be a Arc<dyn PayloadClient> so we can use a mock for testing? we should have some unit tests for the pulling behavior

Reply:
fixed; ut added.